### DISCUSSION TITLE
Revit UnitTest In Rider

### DISCUSSION THREAD

**Discussion by chuongmep** [ricaun-io/RevitTest#8] [Ideas]
Hi @ricaun , that is awesome repo. Thanks for your many repo with Open Source, do we have any challange to make it work in Rider, look like need configuration special in Rider.
![image](https://github.com/ricaun-io/RevitTest/assets/31106432/30d827b9-a3b1-454c-b933-a108ebfa8425)

It works well in Visual Studio, kudo ! 
![image](https://github.com/ricaun-io/RevitTest/assets/31106432/1fc84c37-63b7-4a1e-b83e-39e0740a0c44)

**Comment by ricaun** [comment]
Hello @chuongmep 

I know you need to add some configuration in the Rider to force to use the custom TestAdapter.

This is a really old video that I tested with Rider, and kinda works. [RevitTest in Rider - 1.1.2-beta](https://youtu.be/_xgkd6ItSrg)

Should work with the more recent version, just need to force to use the ricaun.RevitTest.TestAdapter.

**Reply by chuongmep** [reply]
Awesome 😎, Thank you 🙏

**Reply by ricaun** [reply]
I don't have Rider installed anymore, if you could share some screens if you manage to make work with the configurations inside Rider would be great!

**Reply by chuongmep** [reply]
Hi @ricaun  , I think that is some capture I want record if have any working with Rider like me : 
1. Config Unit Test, Enable Vscode Adapter Support.
![rider64_dTtqrB6Qnz](https://github.com/ricaun-io/RevitTest/assets/31106432/b051327a-dbaf-4ab9-b71d-568dc961a9d8)

3. Change Build To Visual Studio Code 
![Y4BSgrd1dy](https://github.com/ricaun-io/RevitTest/assets/31106432/7425326a-68ad-4a2b-a088-06ad8e0c687b)

4. Run Unit Test, some test not pass ,I'm still finding reason 
![rider64_iwNvtLdMQh](https://github.com/ricaun-io/RevitTest/assets/31106432/aa2dcde9-2b2c-400c-9916-d2b88665d1c7)

**Reply by ricaun** [reply]
Nice shots!

I know Rider finds the [NUnit] tests and find the test using the Adapter. So that double the tests, one that runs with the Adapter and another that shows but not run.

In Visual Studio if you clear the project gonna show the [NUnit] tests. And after build, the Adapter gonna execute and show only the tests found with the Adapter.

**Reply by chuongmep** [reply]
I think I know the step 4 to resolve the issue now, from tab Nunit, need change `Metadata` to `TestRunner`, everything will work well under test explore.
![image](https://github.com/ricaun-io/RevitTest/assets/31106432/9b875e0d-1bba-493b-b566-4f7a34950d72)

And here we go ! 
![image](https://github.com/ricaun-io/RevitTest/assets/31106432/11348efc-5248-4197-89dc-88bed080111f)

**Reply by ricaun** [reply]
Neat!
That configuration for the step 4 I didn't find when I was messing with Rider. 
Just found a little odd the step 2, why did you change the MSBuild? I don't remember to change that option, only the step 1.

**Reply by chuongmep** [reply]
Oh @ricaun  I'm using `ARM` Architecture, something Rider Build  default didn't work any more and I need to do it.

**Comment by ricaun** [comment]
Hey @chuongmep,

Could you make some test in the Rider with the `1.9.0-alpha.2` version. That should support `TestFixture` and to fix some weird duplicated test in Visual Studio Test Explorer the only way I found was to change the ExecutorUri to the same as the NUnit executor: `NUnitTestExecutor`

```
<PackageReference Include="ricaun.RevitTest.TestAdapter" Version="1.9.0-alpha.2" />
```

Try this sample code to see if shows correctly inside Rider Test Explorer.

```C#
[TestFixture(1)]
[TestFixture(2)]
[TestFixture(3)]
public class TestsFixture
{
    private readonly int value;
    public TestsFixture(int value)
    {
        this.value = value;
    }

    [Test]
    public void TestValue()
    {
        Console.WriteLine(value);
    }
}
```

Issue reference: #25

**Reply by ricaun** [reply]
That version does not work with Rider. 😒

**Reply by chuongmep** [reply]
Hi @ricaun sorry for my late reply, I'm on vacation, are you considering supported rider ?

**Reply by ricaun** [reply]
No worries, now Rider is free for non commercial. I download and tested.

Yes I rollback the change that break Rider, the version `1.9.0-alpha.3` should work with Rider.

**Reply by chuongmep** [reply]
Wow that was awesome, I will test it after trip. Regards  !

**Comment by paulbuechner** [comment]
Hey, is there a way to get the debugger working in RIDER? Currently, it seems to just run over the set breakpoints...

**Reply by ricaun** [reply]
I don't think there is a way, at least the code only search for the visual studio DTE.
* https://github.com/ricaun-io/ricaun.RevitTest/blob/1.10.1/ricaun.RevitTest.Console/Revit/Utils/VisualStudioDebugUtils.cs

Don't know if there is a way to do the same thing using Rider.
